src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenSkillHandler.java

```diff
@@ -48,7 +48,7 @@
     /**
      * Skill map.
      */
-    public Map<Skill, Tuple<Integer, Double>> skillMap = new HashMap<>();
+    public Map<Skill, Tuple<Integer, Double>> skillMap = new TreeMap<>((s1, s2) -> s1.toString().compareTo(s2.toString()));
```

Comment:
I do not think we should give up hashmap lookups here for a sorting that afaik is not needed here at all

Reply:
GUIs are random sorted based on deserializing order, not sure if creating sorted view every time would be any better

Reply:
ye if its just for a GUI then sort the data in the GUI, aka just sort where needed

Reply:
like we are using that in like three guis, so I don't want to sort them everytime, plus it's just a map of 10 entries
can create sorted map along with this?

Reply:
I'd just put a sorted map there after reading it on client side

Reply:
That entirely depends. How often is this map updated anyway?

Only during into or also during runtime?

And if so how often?

Reply:
Like keys should be populated once per skillHandler instance, since we have fixed amount of skills with fixed code names

Reply:
Will probably create a map view sorted by translations

Reply:
The Tuple is updated fairly often with the xp and level

Reply:
hello, value doesn't change key ordering

Reply:
The key is never updated then.

Reply:
we're just frequently looking it up here, so I see no reason to change to a slower access just for the GUI side

Reply:
Yeah, lookup is very regular. Treemap is log(n) rather than O(1), so for 10, that's like on average double the lookup cost.